packages/watt-pprof-capture/index.js

```diff
@@ -0,0 +1,103 @@
+'use strict'
+
+const pprof = require('@datadog/pprof')
+const {
+  ProfilingNotStartedError,
+  NoProfileAvailableError,
+} = require('./lib/errors')
+const { kITC } = require('../runtime/lib/worker/symbols')
+
+const timeout = (parseInt(process.env.PLT_FLAMEGRAPHS_INTERVAL_SEC) || 60) * 1000
+
+let isCapturing = false // Flag to control capture state
+let latestProfile = null
+let isHandlerRegistered = false
+let captureInterval = null
+let registerTimeout = 10
+let isProfilingInProgress = false // Flag to prevent concurrent pprof calls
+
+function registerHandler () {
+  if (isHandlerRegistered) {
+    return true
+  }
+
+  // Use ITC for handler registration
+  if (globalThis[kITC]) {
+    globalThis[kITC].handle('getLastProfile', getLastProfile)
+    globalThis[kITC].handle('startProfiling', startProfiling)
+    globalThis[kITC].handle('stopProfiling', stopProfiling)
+    isHandlerRegistered = true
+    return true
+  }
+
+  // Keep trying until ITC is available
+  setTimeout(registerHandler, registerTimeout)
```

Comment:
why is this needed?

Reply:
I'm still trying to figure that out, but it seems like `globalThis[kITC]` might not exist yet when this runs, possibly a race condition?